solution/index.js

```diff
@@ -0,0 +1,170 @@
+
+    }
+}
+
+//on mouse out - task in no longer active and keydown event removed
+function mouseOutElement(event) {
+    if(event.target.tagName !== "LI") return;
+    event.target.classList.remove("active");
+    document.removeEventListener("keydown", keyPressMove);
+}
+    
+     
+function search (key) {
+    let liArr = document.querySelectorAll("li"); //get all li elements
+    let query;
+        query = document.getElementById("search").value.toUpperCase();
+    for(let i = 0; i < liArr.length; i++){
+        if(liArr[i].innerText.toLocaleUpperCase().includes(query))
+            liArr[i].hidden = false; //show task if it include search text
+        else
+            liArr[i].hidden = true; //hide task if it doesn't
+        
+    }
+}
+
+//save tasks to API
+async function saveApi() {
+    displayLoading(); //show loading gif
+    let tempTasks = JSON.parse(localStorage.getItem("tasks"));
+    const response = await fetch("https://json-bins.herokuapp.com/bin/614b1c664021ac0e6c080cef", { 
+        method: "PUT",
+        headers:{
+            "Accept": "application/json",
+            "Content-Type": "application/json"
+        },
+        body: JSON.stringify({"tasks": tempTasks})
+    })
+    if(!response.ok){
+        alert("oh no! something went wrong");
+    }
+    document.getElementById("loading").remove(); //remove loading gif
+}
+
+//load tasks from APi
+async function loadApi() {
+    displayLoading(); //show loading gif
+    let liArr = document.querySelectorAll("li");
+    for(let i = 0; i < liArr.length; i++){
+        liArr[i].remove();
+    } 
+    const response = await fetch("https://json-bins.herokuapp.com/bin/614b1c664021ac0e6c080cef",  { 
+        method: "GET",
+        headers:{
+            "Accept": "application/json",
+            "Content-Type": "application/json"
+        }});
+    if(!response.ok){
+        alert("oh no! something went wrong");
+        document.getElementById("loading").remove(); //remove loading gif
+        return;
+    }
+    let tasksI = await response.json();
+    tasks = tasksI.tasks;
+    localStorage.setItem("tasks", JSON.stringify(tasks)); //save to local storage
+    document.getElementById("loading").remove(); //remove loading gif
+    loadTasks(); //display tasks
+}
+
+//display loading gif
+function displayLoading() {
+    let load = document.createElement("div");
+    load.setAttribute("id", "loading");
+    document.body.appendChild(load);
+    load.classList.add("loader");
+    load.classList.add("display");
+}
+
+//empty tasks object
+let tasks = {
+    "todo": [],
+    "in-progress": [],
+    "done": []
+}
+loadTasks();
+document.getElementById("submit-add-to-do").addEventListener("click", handleAddTask);
+document.getElementById("submit-add-in-progress").addEventListener("click", handleAddTask);
+document.getElementById("submit-add-done").addEventListener("click", handleAddTask);
+document.addEventListener("dblclick", handleDBclick);
```

Comment:
You should get specific element by id and add to it event. dont add event to the document itself.